src/formatDate.js

```diff
@@ -50,7 +50,48 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(/\D+/);
+  let formattedDate = '';
+  let year;
+
+  toFormat.forEach((format, i) => {
+    switch (format) {
+      case 'YYYY':
+        if (fromFormat.indexOf('YYYY') !== -1) {
+          year = dateParts[fromFormat.indexOf('YYYY')];
+        } else {
+          year = dateParts[fromFormat.indexOf('YY')];
+        }
+
+        formattedDate += year.length === 2 ? (year < 30) ? '20' + year
+          : '19' + year : year;
```

Comment:
30, 19, 20 are a 'magic number', so make sense to move them to the variables
check other places